### PR TITLE
rename methods/structs/interfaces from Gadget... to BuiltInGadget...

### PR DESCRIPTION
**Pull request by flyth**
This renames methods/structs/interfaces regarding our built-in gadgets to `*BuiltInGadget*` and everything previously named `*OCIGadget*` to only `*Gadget*`. This _will_ break version compatibility between clients/servers in the next release with the previous one (should be okay, as we're still suggest keeping client & server versions in sync anyway).

See also [this](https://github.com/inspektor-gadget/inspektor-gadget/pull/2512#discussion_r1507961453)

### CODE REVIEW

**Review by burak-ok — APPROVED** [review]
LGTM.

> we can later easily remove this renamed part when we deprecate the built-in gadgets

AFAIU yes, that is what we want to do

**Comment by blanquicet** [comment]
@flyth What's the status of this PR?

**Comment by flyth** [comment]
> @flyth What's the status of this PR?

Updated / rebased now.

File: cmd/common/registry.go [hunk 1]
```diff
@@ -255,7 +255,7 @@ func buildCommandFromGadget(
 			onlyArgs := cmd.Flags().Args()
 			if isRunGadget && len(onlyArgs) > 0 {
 				var err error
-				runGadgetInfo, err = runtime.GetGadgetInfo(context.TODO(), gadgetDesc, gadgetParams, onlyArgs)
+				runGadgetInfo, err = runtime.GetBuiltInGadgetInfo(context.TODO(), gadgetDesc, gadgetParams, onlyArgs)
```
[the remark this answers is not in the document]
Comment (flyth) [reply cmd/common/registry.go]: Removed by another PR.

**Comment by flyth** [comment]
Thanks for the reviews!

**Pull request merged**